partner_create_by_vat/models/res_partner.py

```diff
@@ -0,0 +1,160 @@
+                    zip1 = vat_s = state = False
+                    if 'Denumire platitor:' in result.keys():
+                        name = result['Denumire platitor:'].upper()
+                    if 'Adresa:' in result.keys():
+                        adresa = result['Adresa:'].title() or ''
+                    if nrc_key in result.keys():
+                        nrc = result[nrc_key].replace(' ', '')
+                        if nrc == '-/-/-':
+                            nrc = ''
+                    if 'Codul postal:' in result.keys():
+                        zip1 = result['Codul postal:'] or ''
+                    if 'Judetul:' in result.keys():
+                        jud = result['Judetul:'].title() or ''
+                        if jud.lower().startswith('municip'):
+                            jud = ' '.join(jud.split(' ')[1:])
+                        if jud != '':
+                            state = self.env['res.country.state'].search(
+                                [('name', 'ilike', jud)])
+                            if state:
+                                state = state[0].id
+                    if 'Telefon:' in result.keys():
+                        tel = result['Telefon:'].replace('.', '') or ''
+                    if 'Fax:' in result.keys():
+                        fax = result['Fax:'].replace('.', '') or ''
+                    if tva_key in result.keys():
+                        vat_s = bool(
+                            result[tva_key] != 'NU')
+                    self.write({
+                        'name': name or '',
+                        'nrc': nrc or '',
+                        'street': adresa or '',
+                        'phone': tel or '',
+                        'fax': fax or '',
+                        'zip': zip1 or '',
+                        'vat_subjected': vat_s or False,
+                        'state_id': state,
+                    })
+                except:
+                    res = requests.get(
+                        'http://openapi.ro/api/companies/%s.json' % vat_number)
+                    if res.status_code == 200:
+                        res = res.json()
+                        state = False
+                        if res['state']:
+                            state = self.env['res.country.state'].search(
+                                [('name', '=', res['state'].title())])
+                            if state:
+                                state = state[0].id
+                        self.write({
+                            'name': res['name'].upper(),
+                            'nrc': res['registration_id'] and
+                            res['registration_id'].upper() or '',
+                            'street': res['address'].title(),
+                            'city': res['city'].title(),
+                            'phone': res['phone'] and res['phone'] or '',
+                            'fax': res['fax'] and res['fax'] or '',
```

Comment:
You can just say 
```
'fax': res['fax'] or '',
```
Not using res.get('fax') does imply that 'fax' is always present in the dictionary, but may be None or False or so.